### PR TITLE
779 | Configurable details screen

### PR DESCRIPTION
**Pull request by maimoonak**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes required of #779 (except medication request)

**Type**
Choose one: Feature

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/assets/configurations/app/application_workflow.json [hunk 1]
```diff
@@ -44,6 +44,12 @@
         "resource": "PatientRegisterRowViewConfiguration",
         "classification": "patient_register_row",
         "description": "Configurations for Patient Row"
+      },
+      {
+        "workflowPoint": "test_result_detail_view",
+        "resource": "DetailViewConfiguration",
+        "classification": "test_result_detail_view",
+        "description": "Configurations for Test Results Details"
       }
```
Comment (owais-vd) [inline android/quest/src/main/assets/configurations/app/application_workflow.json:53]: I think the above config also needs to add in the quest mapping. cc @ellykits

Comment (maimoonak) [reply android/quest/src/main/assets/configurations/app/application_workflow.json:59]: done!

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt [hunk 2]
```diff
@@ -138,19 +141,35 @@ class QuestPatientDetailActivity : BaseMultiLanguageActivity() {
...
+        if (configurationRegistry.appId == "g6pd") {
+          val encounterId =
+            questionnaireResponse.contained
+              ?.find { it.resourceType == ResourceType.Encounter }
+              ?.logicalId
+          if (encounterId == null) showToast("Missing linked encounter. Invalid data")
+          else
+            startActivity(
+              Intent(this, SimpleDetailsActivity::class.java).apply {
+                putExtra("RECORD_ID", encounterId.replace("#", ""))
```
Comment (owais-vd) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:155]: `RECORD_ID` could get it from constant

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/SimpleDetailsActivity.kt [hunk 3]
```diff
@@ -21,21 +21,24 @@ import androidx.activity.compose.setContent
...
+  val viewModel by viewModels<SimpleDetailsViewModel>()
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    patientId = intent.extras?.getString(QuestionnaireActivity.QUESTIONNAIRE_ARG_PATIENT_KEY) ?: "1"
-    patientViewModel.onBackPressClicked.observe(this, { finish() })
-    patientViewModel.getDemographics(patientId)
-    setContent { AppTheme { QuestPatientTestResultScreen(patientViewModel) } }
+
+    encounterId = intent.extras?.getString("RECORD_ID")!!
```
Comment (owais-vd) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/SimpleDetailsActivity.kt:36]: https://github.com/opensrp/fhircore/pull/936/files#r780226681

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt (hunk 2 above)
Comment (maimoonak) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt]: done

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/SimpleDetailsActivity.kt (hunk 3 above)
Comment (maimoonak) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/SimpleDetailsActivity.kt]: done

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt [hunk 4]
```diff
@@ -138,19 +142,35 @@ class QuestPatientDetailActivity : BaseMultiLanguageActivity() {
...
-            .putExtras(
-              QuestionnaireActivity.intentArgs(
-                clientIdentifier = patientId,
-                formName = questionnaireId,
-                readOnly = true,
-                populationResources = populationResources
-              )
+        // TODO https://github.com/opensrp/fhircore/issues/778
+        //  1- handle via config/menu, 2- pass encounterId from quest details screen
+        if (configurationRegistry.appId == "g6pd") {
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:147]: k, so 778 is going to remove this app specific check? can you add a quick not about that in #778, it's clear from reading that issue that replacing this with something generic is required for it

Comment (owais-vd) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:147]: the above implementation has been removed here https://github.com/opensrp/fhircore/blob/70f47f52518ae4008c5681738204568e31e24819/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt#L159-L161

Comment (pld) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:147]: cool, is that on a branch?

**Review by pld — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt (hunk 4 above)
Comment (maimoonak) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:147]: done

Comment (owais-vd) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt:147]: yes

**Pull request merged**